rust/src/io/object_store.rs

```diff
@@ -54,21 +57,72 @@ impl std::fmt::Display for ObjectStore {
     }
 }
 
+const AWS_CREDS_CACHE_KEY: &str = "aws_credentials";
 #[derive(Debug)]
 struct AwsCredentialAdapter {
-    pub inner: DefaultCredentialsChain,
+    pub inner: Arc<dyn ProvideCredentials>,
+
+    // RefCell can't be shared accross threads, so we use HashMap
+    cache: Arc<RwLock<HashMap<String, Arc<aws_credential_types::Credentials>>>>,
```

Comment:
You mean it is just one key `AWS_CREDS_CACHE_KEY` in this map?

Curious whether just use `Option` ?

Reply:
We can't change the value inside Option. compiler doesn't let me do`tokio::sync::RwLock<RefCell>` either claiming that it's not safe for multi-thread